### ISSUE TITLE
Rule Request: Untyped error in catch statement

### ISSUE THREAD

**Issue by dirtydanee** [realm/SwiftLint#2045]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

#### Description

When an `error` is being thrown from a `do - catch`, the type of the error being thrown is unknown. We might would like to handle some specific types of errors in a specific way, but most of the times the `error` is being logged to the console or reported to some third party error tracking system. In order to do that, we do not need to unwrap the `error` without the type itself, it is available for us in the `catch` statement.

The official [Apple documentation](https://developer.apple.com/library/content/documentation/Swift/Conceptual/Swift_Programming_Language/ErrorHandling.html) on error handling is without a single example of unwrapping an error without specifying the type. 

#### What would trigger?
```swift
enum Error: Swift.Error {
    case invalidOperation
    case failedOperation
}

func foo() throws {
    throw Error.invalidOperation
}

do {
    try foo()
} catch let error {
    print("Action failed. Error: \(error.localizedDescription)")
}
```
#### What would not trigger?
```swift
enum Error: Swift.Error {
    case invalidOperation
    case failedOperation
}

func foo() throws {
    throw Error.invalidOperation
}
```
Example 1:

```swift
do {
    try foo()
} catch Error.invalidOperation {
    print("Action failed with invalid operation")
} catch Error.failedOperation {
    print("Action failed with failed operation")
} catch {
    print("Action failed for unkown reason. Error: \(error.localizedDescription)")
}
```
Example 2:
```swift
do {
    try foo()
}  catch let error as Error {
   print("Action failed with invalid operation")
} catch {
    print("Action failed for unknown reason. Error: \(error.localizedDescription)")
}
```
Example 3:
```swift
do {
    try foo()
} catch Error.invalidOperation {
   print("Action failed with invalid operation")
}  catch let error as Error {
   print("Action failed with custom error")
} catch {
    print("Action failed for unknown reason. Error: \(error.localizedDescription)")
}
```
#### Configurable?
The rule should not be configurable. 

#### Opt-in or enabled?
This rule could be a personal preference or coding style from various individuals, so it could be kept as opt-in.

**Comment by marcelofabri** [comment]
Implemented in #2045 and #2101
